src/token/tests.rs

```diff
@@ -84,7 +84,7 @@ fn example_ec_key_two(alg: Algorithm) -> CoseKey {
 fn example_headers(alg: Algorithm) -> (Header, Header) {
     let unprotected_header = HeaderBuilder::new()
         .iv(vec![
-            0x63, 0x68, 0x98, 0x99, 0x4F, 0xF0, 0xEC, 0x7B, 0xFC, 0xF6, 0xD3, 0xF9, 0x5B,
+            0x63, 0x68, 0x98, 0x99, 0x4F, 0xF0, 0xEC, 0x7B, 0xFC, 0xF6, 0xD3, 0xF9,
```

Comment:
The IV length is fixed for any given algorithm.

The current examples use AES-GCM, which has a fixed IV/nonce size of 12, and having more or less bytes in the header than the algorithm requires will now cause an error due to the additional checks I have added.

Now that I think about it, it would probably be better to use `aes_algorithm_iv_length()` here and either randomly generate or truncate the IV to the right size.